lmd/datarow.go

```diff
@@ -746,3 +746,171 @@ func (d *DataRow) WriteJSON(json *jsoniter.Stream, columns *[]*Column) {
 	}
 	json.WriteRaw("]\n")
 }
+
+func (d *DataRow) isAuthorizedFor(authUser string, host string, service string) (canView bool) {
+	canView = false
+
+	p := d.DataStore.Peer
+
+	// get contacts for host, if we are checking a host or
+	// if this is a service and ServiceAuthorization is loose
+	if (service != "" && p.LocalConfig.ServiceAuthorization == loose) || service == "" {
+		hostObj, ok := p.Tables["hosts"].Index[host]
+		contactsColumn := p.Tables["hosts"].GetColumn("contacts")
+		// Make sure the host we found is actually valid
+		if ok {
```

Comment:
what about
```
if !ok {
return
}
```
This would save one level of indentation.

Reply:
Theoretically I guess it could happen that the host lookup for some reason fails, but the service lookup later is ok, and the service contains the searched for contact.

Although these object lookups probably shouldn't fail ?

Reply:
this should not happen. lmd uses locking to always provide a consistent set of hosts and services.

Reply:
Done. Perhaps we could skip the OK checks altogether, but probably a good safeguard to have.